### ISSUE TITLE
Add zippable extensions configurable

### ISSUE THREAD

**Issue by hgomez** [dependency-check/DependencyCheck#65]
Some internal projects are using their own extension, ".nar" which is a jar file with jars / classes inside (OSGI like).

EXTENSIONS or ZIPPABLES in ArchiveAnalyzer should have a way to add these custom/inhouse file extensions

**Comment by jeremylong** [comment]
NAR is not exactly a custom extension. But I see your point... I will add .NAR to the list of zip archives until I find time to make this list cusomizable (part of the issue with customizable is making it flexible and easy to use so that we can support file extension to compression method - I'll have to think about this).

**Comment by hgomez** [comment]
Yep, .nar is an internal usage but there will be more elsewhere.
That's why I provided code to support it cleanly, in cli, ant and maven (see https://github.com/jeremylong/DependencyCheck/pull/63)

GitHub get my 2 changes (disable analysis for Maven Provides/Runtime scope) and extra extensions into a single PR, my bad ;(

**Comment by jeremylong** [comment]
I just started merging your PR. I am going to update a few things, rename the parameters slightly, etc.  There was a problem with the implementation in that the configured extensions would be analyzed by the ArchiveAnalyzer - but because it wasn't one of the hard coded extensions it would not extract the contents...  I made the configuration option zipExtensions instead of extraExtensions. Partially so we could more easily add support for other archive types using non-standard extensions.

I should have the code updated and pushed within a day or two.

--Jeremy
